src/model/about/aboutfriend.cpp

```diff
@@ -105,8 +105,20 @@ bool AboutFriend::clearHistory()
     History* const history = Nexus::getProfile()->getHistory();
     if (history) {
         history->removeFriendHistory(pk.toString());
+        f->emitChatHistoryErasedSignal();
```

Comment:
1. No one should manage signals of external class. I.e. it's not good practice to do `emit var.signal()` or `var.emitSignal()` where `emitSignal` just emit signal inside a `var`. Signal emitting should be result of some action or result of reaction on some action.
2. Also in current qTox architecture `Friend` is more like struct with information about Tox friend, than class with behaviour.

I'm not sure, what is the best way. But I have an idea. Add signal in `History` (something like `friendHistoryCleared(Friend*)`) and connect in `ChatForm` constructor. Don't forget to check `Friend` equals to friend stored in `ChatForm`.